go/vt/vtgate/vstream_manager.go

```diff
@@ -408,6 +411,21 @@ func (vs *vstream) alignStreams(ctx context.Context, event *binlogdatapb.VEvent,
 	}
 }
 
+func (vs *vstream) getCells() []string {
+	var cells []string
+	if vs.optCells != "" {
+		for _, cell := range strings.Split(strings.TrimSpace(vs.optCells), ",") {
+			cells = append(cells, strings.TrimSpace(cell))
+		}
+	}
+
+	vtgateCell := vs.vsm.cell
```

Comment:
Not sure if this is a pointer or not, but any reason not to move this into the conditional and even remove it entirely?
```
 	if len(cells) == 0 {
                # use the vtgate's cell by default
 		return []string{vs.vsm.cell}
 	}
```

So when combined with the previous comment, the function could be shortened to:
```
if cells := strings.Split(strings.TrimSpace(vs.optCells), ","); len(cells) != 0 {
    return cells
} else {
    # use the vtgate's cell by default
    return []string{vs.vsm.cell}
}
```

Nit, but IMO that's simpler for humans and the runtime.

Reply:
removed the intermediate variable and replaced with suggested comment.